### PR TITLE
CLI-1398: Update CLI references to v2

### PR DESCRIPTION
**Pull request by brianstrauch**
### Description 

Update `confluent` v1 / `ccloud` syntax to `confluent` v2 syntax.

### Author Validation

_Describe the validation already done, or needs to be done, by the PR submitter._

- [X] Documentation 
- [x] Run cp-demo

- [ ] **REMINDER: merge simultaneously with https://github.com/confluentinc/examples/pull/1047**

### Reviewer Tasks

_Describe the tasks/validation that the PR submitter is requesting to be done by the reviewer._

### CODE REVIEW

**Comment by ybyzek** [comment]
@chuck-confluent is there going to be a new `tools` Docker image with the updated CLIs, and how will it provide the unified CLI?

**Comment by ybyzek** [comment]
> @chuck-confluent is there going to be a new `tools` Docker image with the updated CLIs, and how will it provide the unified CLI?

Correction: cp-demo is no longer using training/tools image, it is now using https://hub.docker.com/r/cnfldemos/tools which is built from: https://github.com/confluentinc/cp-demo/blob/7.0.0-post/tools/Dockerfile-tools .  So @brianstrauch this PR should also modify that Dockerfile and we in DevX can validate and publish the updated Docker image

**Review by DABH — APPROVED** [review]
All looks good to me so far 👍

**Review by ybyzek — CHANGES_REQUESTED** [review]
Even though in Draft, marking `Request changes` just to ensure Dockerfile is updated and PR is validated

File: tools/Dockerfile-tools
Comment (ybyzek) [inline tools/Dockerfile-tools:17]: If CP 7.0.0 is supported only with confluent v1, then it needs to include confluent v1 for on-prem and confluent v2 for ccloud 

cc: @ethanruhe @DABH

File: docs/hybrid-cloud.rst
Comment (ybyzek) [inline docs/hybrid-cloud.rst:206]: This is using v2, but only v1 is supported with CP 7.0?

cc: @ethanruhe @DABH

**Review by ybyzek — CHANGES_REQUESTED** [review]

File: docs/hybrid-cloud.rst
Comment (ybyzek) [inline docs/hybrid-cloud.rst:43]:
```suggestion
#. Install `Confluent CLI <https://docs.confluent.io/confluent-cli/current/install.html>`__ v2.0.0 or later to manage |ccloud|.  Do not confuse this Confluent CLI binary v2 with the Confluent CLI binary v1 that is used to managed on-prem |cp| deployments.  See `documentation <https://docs.confluent.io/confluent-cli/current/migrate.html>`__ for more information on the CLI migration and running the CLIs in parallel.
```

**Comment by ybyzek** [comment]
Because cp-demo hybrid module uses `ccloud-stack`, this PR is blocked by https://github.com/confluentinc/examples/pull/1047 (actually they should be merged simultaneously)

File: scripts/helper/functions.sh
Comment (ybyzek) [inline scripts/helper/functions.sh:280]: @brianstrauch I looked into this issue, and it works with `-vvvv` but fails without `-vvvv`.   This is a regression and results in a lot of debug output that is distracting to the demo.  Any ideas on what's happening here?

Comment (brianstrauch) [reply scripts/helper/functions.sh:280]: In one of the later 1.x versions of the CLI (somewhere around 1.30 iirc) we started hiding the "login success" message (not sure why that was, this happened before I joined). I can experiment with less verbosity (maybe `-vv` or `-vvv`) so we can strike a good balance where the demo works but the output isn't distracting.

Comment (brianstrauch) [reply scripts/helper/functions.sh:280]: Unfortunately `-vv` prints nothing and `-vvv` prints too much. How about we print a generic "Logged in to the Confluent CLI" message if login succeeds, and the full output only if it fails?

Comment (ybyzek) [reply scripts/helper/functions.sh:280]: It should work without any verbose flags. Perhaps the script can be modified to check error codes:

```
root@tools:/# confluent-v1 login --ca-cert-path /etc/kafka/secrets/snakeoil-ca-1.crt --url $MDS_URL
Enter your Confluent credentials:
Username: kjdf
Password: ****
Error: metadata service backend error: 401 Unauthorized: {
"servlet":"default",
"message":"Unauthorized",
"url":"/security/1.0/authenticate",
"status":"401"
}
root@tools:/# echo $?
1
root@tools:/# confluent-v1 login --ca-cert-path /etc/kafka/secrets/snakeoil-ca-1.crt --url $MDS_URL
Enter your Confluent credentials:
Username: superUser
Password: *********
root@tools:/# echo $?
0
```

Comment (ybyzek) [reply scripts/helper/functions.sh:280]: Or search for errors in output

Comment (ybyzek) [reply scripts/helper/functions.sh:280]: @brianstrauch WDYT about this solution?


```
[12:55:51] ~/git/cp-demo/scripts(PR-404) ✗: git diff
diff --git a/scripts/helper/functions.sh b/scripts/helper/functions.sh
index 7811c4e..b2ad77e 100755
--- a/scripts/helper/functions.sh
+++ b/scripts/helper/functions.sh
@@ -281,12 +281,13 @@ mds_login()
     expect "Username: "
     send "${SUPER_USER}\r";
     expect "Password: "
+    log_user 0
     send "${SUPER_USER_PASSWORD}\r";
+    log_user 1
     expect "Logged in as "
     set result $expect_out(buffer)
 END
   )
-  echo "$OUTPUT"
   if [[ ! "$OUTPUT" =~ "Logged in as" ]]; then
     echo "Failed to log into MDS. Please check all parameters and run again."
     exit 1
```

**Review by ybyzek — COMMENTED** [review]
@brianstrauch does this issue happen on your end as well?

```
root@tools:/# which confluent-v1
root@tools:/# which confluent
/usr/bin/confluent
root@tools:/# 
```

**Comment by brianstrauch** [comment]
> @brianstrauch does this issue happen on your end as well?

@ybyzek I would expect to see `/usr/bin/confluent-v1` as the output for `which confluent-v1`, if that code snippet is from a Docker container.

File: scripts/helper/functions.sh
Comment (brianstrauch) [reply scripts/helper/functions.sh:280]: My favorite solution of the two is to check for "Error:" in the output.

Comment (brianstrauch) [reply scripts/helper/functions.sh:280]: Checking for error codes seems to be the cleanest approach here!

Comment (ybyzek) [reply scripts/helper/functions.sh:280]: @brianstrauch can you please validate the code below (it works in my env) and then drop/replace into your PR?

```
mds_login()
{
  MDS_URL=$1
  SUPER_USER=$2
  SUPER_USER_PASSWORD=$3

  # Log into MDS
  if [[ $(type expect 2>&1) =~ "not found" ]]; then
    echo "'expect' is not found. Install 'expect' and try again"
    exit 1
  fi
  echo -e "\n# Login to MDS using Confluent CLI"
  OUTPUT=$(
  expect <<END
    log_user 1
    spawn confluent-v1 login --ca-cert-path /etc/kafka/secrets/snakeoil-ca-1.crt --url $MDS_URL
    expect "Username: "
    send "${SUPER_USER}\r";
    expect "Password: "
    send "${SUPER_USER_PASSWORD}\r";
    expect EOF
    catch wait result
    exit [lindex \$result 3]
END
  )
  if [ $? -ne 0 ]; then
    echo "Failed to log into MDS. Please check all parameters and run again."
    exit
  fi
}
```

**Comment by ybyzek** [comment]
@brianstrauch https://hub.docker.com/layers/cnfldemos/tools/0.3/images/sha256-90e9983e63d519becd20878249b0448c67f32c4381b5967620e3a3291c85347e?context=explore (tag 0.3) is now published on Docker Hub.  Can you please update the docker-compose.yml file as part of the PR?  https://github.com/confluentinc/cp-demo/blob/7.0.0-post/docker-compose.yml#L52

**Comment by ybyzek** [comment]
@brianstrauch bump on https://github.com/confluentinc/cp-demo/pull/404#discussion_r753160486 (which might be one of the last changes required to unblock script execution)

File: scripts/helper/functions.sh
Comment (brianstrauch) [reply scripts/helper/functions.sh]: Worked perfectly- updated!

**Comment by ybyzek** [comment]
I have verified the CCloud portion of the demo with the new `ccloud_library.sh` file in https://github.com/confluentinc/examples/pull/1047

**Review by ybyzek — APPROVED** [review]
@brianstrauch LGTM, and THANK YOU ✨ !

Can you please merge these 2 PRs simultaneously?

https://github.com/confluentinc/examples/pull/1047
https://github.com/confluentinc/cp-demo/pull/404

**Pull request merged**
